x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -34,77 +27,23 @@ function updateEnvironmentUrl(
   });
 }
 
-const SEPARATOR_OPTION = {
-  text: `- ${i18n.translate(
-    'xpack.apm.filter.environment.selectEnvironmentLabel',
-    { defaultMessage: 'Select environment' }
-  )} -`,
-  disabled: true,
-};
-
-function getOptions(environments: string[]) {
-  const environmentOptions = environments
-    .filter((env) => env !== ENVIRONMENT_NOT_DEFINED.value)
-    .map((environment) => ({
-      value: environment,
-      text: environment,
-    }));
-
-  return [
-    ENVIRONMENT_ALL_SELECT_OPTION,
-    ...(environments.includes(ENVIRONMENT_NOT_DEFINED.value)
-      ? [ENVIRONMENT_NOT_DEFINED]
-      : []),
-    ...(environmentOptions.length > 0 ? [SEPARATOR_OPTION] : []),
-    ...environmentOptions,
-  ];
-}
-
 export function ApmEnvironmentFilter() {
-  const { status, environments, environment } = useEnvironmentsContext();
-
-  return (
-    <EnvironmentFilter
-      status={status}
-      environment={environment}
-      environments={environments}
-    />
-  );
-}
-
-export function EnvironmentFilter({
-  environment,
-  environments,
-  status,
-}: {
-  environment: Environment;
-  environments: Environment[];
-  status: FETCH_STATUS;
-}) {
+  const { environment, serviceName, environmentOptions, status, start, end } =
+    useEnvironmentsContext();
```

Comment:
I think I mentioned this before, but I can no longer find it - I don't think `useEnvironmentsContext` should return all this information. I don't see a compelling reason for `serviceName`, `environmentOptions` and `start` and `end` to be there. It's better for `useEnvironmentsContext` to only return what is necessary, to separate concerns.